### PR TITLE
Wait for images to be ready before moving between releases

### PR DESCRIPTION
**Pull request by pipex**
For download-then-kill strategy, this waits for all changing images on the target
release to be available on device before killing the old services. This
will prevent that multicontainer applications get to a state where some
services of the new release start runnning much before others have been
downloaded.

When adding new services to a multicontainer app, the supervisor will
now wait for other changing services to be downloaded before starting
the new service.

Closes: #1812
Change-type: patch

### CODE REVIEW

**Comment by cywang117** [comment]
LGTM!

**Comment by pipex** [comment]
@balena-ci retest

**Comment by pipex** [comment]
@balena-ci retest

**Comment by pipex** [comment]
@balena-ci rebase

File: src/compose/app.ts [hunk 1]
```diff
@@ -664,16 +674,16 @@ export class App {
...
 	}
 
 	// Unless the update strategy requires an early kill (i.e kill-then-download,
 	// delete-then-download), we only want to kill a service once the images for the
 	// services it depends on have been downloaded, so as to minimize downtime (but not
-	// block the killing too much, potentially causing a daedlock)
+	// block the killing too much, potentially causing a deadlock)
 	private dependenciesMetForServiceKill(
-		target: Service,
+		_target: Service,
```
Comment (20k-ultra) [inline src/compose/app.ts:686]: Remove this param and the 1 other reference that passes it

**Comment by pipex** [comment]
@balena-ci I self certify!

**Pull request merged**